ui/app/adapters/sync/destination.js

```diff
@@ -4,32 +4,25 @@
  */
 
 import ApplicationAdapter from 'vault/adapters/application';
+import { pluralize } from 'ember-inflector';
 
 export default class SyncDestinationAdapter extends ApplicationAdapter {
-  namespace = 'v1';
+  namespace = 'v1/sys';
 
-  _baseUrl() {
-    return `${this.buildURL()}/sys`;
+  pathForType(modelName) {
+    return modelName === 'sync/destination' ? pluralize(modelName) : modelName;
   }
 
-  createRecord(store, type, snapshot) {
-    const { name, type: destinationType } = snapshot.attributes();
-    const url = `${this._baseUrl()}/sync/destinations/${destinationType}/${name}`;
-
-    return this.ajax(url, 'POST', { data: snapshot.serialize() }).then((resp) => ({
-      id: `${destinationType}/${name}`,
+  createRecord(store, { modelName }, snapshot) {
+    const { name, type } = snapshot.attributes();
+    const data = snapshot.serialize();
+    return this.ajax(this.buildURL(modelName, name), 'POST', { data }).then((resp) => ({
+      id: `${type}/${name}`,
       ...resp,
     }));
   }
 
-  // modelName is sync/destinations/:type
-  // id is the destination name
-  urlForFindRecord(id, modelName) {
```

Comment:
I noticed this error in the console now that this is removed 

```
You requested a record of type 'sync/destinations/aws-sm' with id 'dest-3' but the adapter returned a payload with primary data having an id of 'aws-sm/dest-3'. Use 'store.findRecord()' when the requested id is the same as the one returned by the adapter. In other cases use 'store.queryRecord()' instead.
```

if you want to remove this, then we should probably update the details router to use `queryRecord()`

Reply:
Thanks for pointing that out and testing! I see that it was coming back as a warning since the format of the id was being changed in the serializer. I've made some updates to use the name as the id for destination child models which call `findRecord` and left the `type/name` formatting as is in `extractLazyPaginatedData` used by the base model.